Account/FirefoxAccount.swift

```diff
@@ -349,6 +369,15 @@ open class FirefoxAccount {
         return client.destroyDevice(ownDeviceId: ownDeviceId, withSessionToken: session.sessionToken as NSData) >>> succeed
     }
 
+    fileprivate func registerOrUpdateDevice(session: TokenState, force: Bool = false) -> Success {
+        return FxADeviceRegistrator.registerOrUpdateDevice(self, sessionToken: session.sessionToken as NSData, force: force).bind { result in
```

Comment:
If this fails, what happens? Are we out of sync with our device registration? Will we try again?

Reply:
Thanks, I added https://github.com/mozilla-mobile/firefox-ios/pull/3491/files#diff-e7993812e8ded083c5d2b13f1f46dc18R295 which does the same as Android.